rueidiscompat/adapter.go

```diff
@@ -4835,6 +4835,615 @@ func (c *Compat) Watch(ctx context.Context, fn func(Tx) error, keys ...string) e
+		}
+		// [NOFREQS]
+		if options.NoFreqs {
+			_cmd = cmds.Incomplete(cmds.FtCreateIndex(_cmd).Nofreqs())
+		}
+		// [STOPWORDS count [stopword ...]]
+		if len(options.StopWords) > 0 {
+			_cmd = cmds.Incomplete(cmds.FtCreateIndex(_cmd).Stopwords(int64(len(options.StopWords))).Stopword(argsToSlice(options.StopWords)...))
+		}
+		// [SKIPINITIALSCAN]
+		if options.SkipInitialScan {
+			_cmd = cmds.Incomplete(cmds.FtCreateIndex(_cmd).Skipinitialscan())
+		}
+	}
+	_cmd = cmds.Incomplete(cmds.FtCreateIndex(_cmd).Schema())
+	// 	SCHEMA field_name [AS alias] TEXT | TAG | NUMERIC | GEO | VECTOR | GEOSHAPE [ SORTABLE [UNF]]
+	//   [NOINDEX] [ field_name [AS alias] TEXT | TAG | NUMERIC | GEO | VECTOR | GEOSHAPE [ SORTABLE [UNF]] [NOINDEX] ...]
+	for _, sc := range schema {
+		if sc.FieldName == "" || sc.FieldType == SearchFieldTypeInvalid {
+			panic("FT.CREATE: SCHEMA FieldName and FieldType are required")
+		}
+		_cmd = cmds.Incomplete(cmds.FtCreateSchema(_cmd).FieldName(sc.FieldName))
+		if sc.As != "" {
+			_cmd = cmds.Incomplete(cmds.FtCreateFieldFieldName(_cmd).As(sc.As))
+		}
+		switch sc.FieldType {
+		case SearchFieldTypeNumeric:
+			_cmd = cmds.Incomplete(cmds.FtCreateFieldAs(_cmd).Numeric())
+		case SearchFieldTypeTag:
+			_cmd = cmds.Incomplete(cmds.FtCreateFieldAs(_cmd).Tag())
+		case SearchFieldTypeText:
+			_cmd = cmds.Incomplete(cmds.FtCreateFieldAs(_cmd).Text())
+		case SearchFieldTypeGeo:
+			_cmd = cmds.Incomplete(cmds.FtCreateFieldAs(_cmd).Geo())
+		case SearchFieldTypeVector:
+			// FIXME: implement this
+			// VECTOR <algorithm> <index_attribute_count> <index_attribute_name> <index_attribute_value>
+			// [<index_attribute_name> <index_attribute_value> ...]
+			// Ref: https://redis.io/docs/latest/develop/interact/search-and-query/advanced-concepts/vectors/#create-a-vector-index
+			if sc.VectorArgs == nil {
+				panic("FT.CREATE: SCHEMA VectorArgs cannot be nil")
+			}
+			if sc.VectorArgs.FlatOptions != nil && sc.VectorArgs.HNSWOptions != nil {
+				panic("FT.CREATE: SCHEMA VectorArgs FlatOptions and HNSWOptions are mutually exclusive")
+			}
+			if sc.VectorArgs.FlatOptions != nil {
+				// args = append(args, "FLAT")
+				// _cmd = cmds.Incomplete(cmds.FtCreateFieldFieldTypeVector(_cmd).())
+				if sc.VectorArgs.FlatOptions.Type == "" || sc.VectorArgs.FlatOptions.Dim == 0 || sc.VectorArgs.FlatOptions.DistanceMetric == "" {
+					panic("FT.CREATE: Type, Dim and DistanceMetric are required for VECTOR FLAT")
+				}
+				flatArgs := []interface{}{
+					"TYPE", sc.VectorArgs.FlatOptions.Type,
+					"DIM", sc.VectorArgs.FlatOptions.Dim,
+					"DISTANCE_METRIC", sc.VectorArgs.FlatOptions.DistanceMetric,
+				}
+				if sc.VectorArgs.FlatOptions.InitialCapacity > 0 {
+					flatArgs = append(flatArgs, "INITIAL_CAP", sc.VectorArgs.FlatOptions.InitialCapacity)
+				}
+				if sc.VectorArgs.FlatOptions.BlockSize > 0 {
+					flatArgs = append(flatArgs, "BLOCK_SIZE", sc.VectorArgs.FlatOptions.BlockSize)
```

Comment:
Lint Error: adapter.go:5132:6: this value of flatArgs is never used (SA4006)

Reply:
fixed.